packages/babel-parser/test/fixtures/typescript/cast/as/output.json

```diff
@@ -236,7 +236,7 @@
           }
         },
         "expression": {
-          "type": "BinaryExpression",
+          "type": "TSAsExpression",
```

Comment:
This change is incorrect.

```js
// input
x === 1 as number;

// Before
x === (1 as number);

// After
(x === 1) as number;
```

It's caused by the  `in` operator having a difference precedence than the relational operators. We should add a test case that `types. _in.binop === types.relational.binop`.

Reply:
Understood. Fixing the error and adding a test case now.